idaes/power_generation/properties/tests/test_NG_PR.py

```diff
@@ -0,0 +1,172 @@
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+
+import pytest
+
+from idaes.core import FlowsheetBlock
+from idaes.generic_models.properties.core.eos.ceos import \
+    cubic_roots_available
+from idaes.power_generation.properties.natural_gas_PR import get_NG_properties
+from idaes.generic_models.properties.core.generic.generic_property import (
+        GenericParameterBlock)
+from idaes.generic_models.unit_models import GibbsReactor
+from pyomo.util.check_units import assert_units_consistent
+
+from pyomo.environ import (ConcreteModel,
+                           Objective,
+                           SolverFactory,
+                           SolverStatus,
+                           TerminationCondition,
+                           value)
+
+from idaes.core.util.testing import get_default_solver
+
+import idaes.logger as idaeslog
+SOUT = idaeslog.INFO
+
+# Set module level pyest marker
+pytestmark = pytest.mark.cubic_root
+prop_available = cubic_roots_available()
+
+# Get default solver for testing
+solver = get_default_solver()
+
+
+class TestNaturalGasProps(object):
+    @pytest.fixture()
+    def m(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={'dynamic': False})
+
+        m.fs.props = GenericParameterBlock(default=get_NG_properties())
+        m.fs.state = m.fs.props.build_state_block(
+                [1],
+                default={"defined_state": True})
+
+        return m
+
+    @pytest.mark.integration
+    def test_T_sweep(self, m):
+        assert_units_consistent(m)
+
+        m.fs.obj = Objective(expr=(m.fs.state[1].temperature - 510)**2)
+
+        for logP in range(8, 13, 1):
+            m.fs.obj.deactivate()
+
+            m.fs.state[1].flow_mol.fix(100)
+            m.fs.state[1].mole_frac_comp["H2"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["CO"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["H2O"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["CO2"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["O2"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["N2"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["Ar"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["CH4"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["C2H6"].fix(0.1)
+            m.fs.state[1].mole_frac_comp["C3H8"].fix(0.05)
+            m.fs.state[1].mole_frac_comp["C4H10"].fix(0.05)
+            m.fs.state[1].temperature.fix(300)
+            m.fs.state[1].pressure.fix(10**(0.5*logP))
+
+            m.fs.state.initialize(outlvl=0)
+
+            m.fs.state[1].temperature.unfix()
+            m.fs.obj.activate()
+
+            solver = SolverFactory('ipopt')
```

Comment:
Create the solver before doing the for loop - otherwise you are just repeating this step for no reason.